terraform/modules/spack/sentry.tf

```diff
@@ -0,0 +1,207 @@
+  yaml_body = <<-YAML
+    kind: Role
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: create-sentry-api-key
+      namespace: sentry
+    rules:
+      - apiGroups: ["", "extensions", "apps"]
+        resources: ["deployments", "pods", "pods/exec"]
+        verbs: ["get", "list", "create"]
+  YAML
+}
+
+resource "kubectl_manifest" "sentry_api_key_rolebinding" {
+  yaml_body = <<-YAML
+    kind: RoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+      name: create-sentry-api-key
+      namespace: sentry
+    subjects:
+      - kind: ServiceAccount
+        name: ${kubectl_manifest.sentry_api_key_serviceaccount.name}
+    roleRef:
+      kind: Role
+      name: ${kubectl_manifest.sentry_api_key_role.name}
+      apiGroup: rbac.authorization.k8s.io
+  YAML
+}
+
+locals {
+  spackbot_email = "mike.vandenburgh+spackbot@kitware.com"
+  permissions = [
+    "alerts:read",
+    "alerts:write",
+    "event:admin",
+    "event:read",
+    "event:write",
+    "member:admin",
+    "member:read",
+    "member:write",
+    "org:admin",
+    "org:integrations",
+    "org:read",
+    "org:write",
+    "project:admin",
+    "project:read",
+    "project:releases",
+    "project:write",
+    "team:admin",
+    "team:read",
+    "team:write",
+  ]
+}
+
+locals {
+  api_key_scopes = "[${join(",", [
+    for permission in local.permissions : "'${permission}'"
+  ])}]"
+}
+
+resource "kubernetes_job" "sentry_api_key_job" {
+  depends_on = [
+    kubectl_manifest.sentry_api_key_rolebinding,
+  ]
+  metadata {
+    name      = "create-sentry-api-key-job"
+    namespace = "sentry"
+  }
+  spec {
+    completions = 1
+    template {
+      metadata {}
+      spec {
+        container {
+          name              = "create-sentry-api-key"
+          image             = "bitnami/kubectl"
+          image_pull_policy = "IfNotPresent"
+          command           = ["/bin/sh", "-c"]
+          args = [
+            <<HEREDOC
+            kubectl -n sentry exec -i deploy/sentry-web -- sentry createuser --no-input --force-update --email='${local.spackbot_email}' --password='${random_password.sentry_spackbot_password.result}' --superuser &&
+            kubectl -n sentry exec -i deploy/sentry-web -- sentry shell -c "from sentry.models import ApiToken, User; ApiToken.objects.update_or_create(user=User.objects.get(username='${local.spackbot_email}'), token='${random_password.sentry_api_key.result}', defaults=dict(scopes=0, scope_list=${local.api_key_scopes}))";
+            HEREDOC
+          ]
+        }
+        restart_policy       = "OnFailure"
+        service_account_name = kubectl_manifest.sentry_api_key_serviceaccount.name
+        node_selector = {
+          "spack.io/node-pool" = "base"
+        }
+      }
+    }
+  }
+  # Setting this to true forces Terraform to wait for the API
+  # key to be created before creating any sentry resources.
+  wait_for_completion = true
+}
```

Comment:
Any thoughts on this approach @AlmightyYakob @danlamanna ? This lets us avoid the chicken-and-egg problem that comes with using the Sentry terraform provider. For reference, I got the Python code from https://github.com/getsentry/sentry/issues/46911

Reply:
Works for me 🤷‍♂️

Reply:
>  This lets us avoid the chicken-and-egg problem that comes with using the Sentry terraform provider.

What is the exact chicken-and-egg problem? Is it that you need to have users or api tokens created before you can use sentry, yet we're trying to spin up sentry from scratch?

Reply:
The former.

Reply:
Well, to be more specific - you need a user/API key in order to configure Sentry with the Terraform provider. But to create an API key, you need to create a user on Sentry. Ideally, Terraform takes care of provisioning the API key, and then feeds that into the sentry provider.